source/extensions/tracers/opentelemetry/resource_detectors/resource_detector.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "envoy/config/typed_config.h"
+#include "envoy/server/tracer_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+using ResourceAttributes = std::map<std::string, std::string>;
+
+struct Resource {
```

Comment:
nit: please add a short comment to describe what this represents.

Reply:
Done in [a374106](https://github.com/envoyproxy/envoy/pull/29547/commits/a374106c23c6f855cd0bdba275707c3449d20118)